sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/SqlTransform.java

```diff
@@ -87,6 +88,10 @@
 
   abstract String queryString();
 
+  abstract Object queryParameters();
```

Comment:
My preference would be to have slightly more typed `positionalParameters` and `namedParameters`. If you like, you can do it robustly with `@AutoOneOf`, which would subsume also `queryParameterMode`.

https://github.com/google/auto/blob/master/value/userguide/howto.md#-make-a-class-where-only-one-of-its-properties-is-ever-set

Reply:
Will do, thanks for the tip.

Reply:
Looks like we will have to update to AutoValue 1.7 to be able to do none (void is not allowed in version 1.6.3 that we are using)